gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -193,7 +197,18 @@ protected void processMessage(DecodeableKafkaRecord<byte[],byte[]> message) {
         org.apache.gobblin.configuration.State jobStatus = parseJobStatus(gobblinTrackingEvent);
         if (jobStatus != null) {
           try (Timer.Context context = getMetricContext().timer(GET_AND_SET_JOB_STATUS).time()) {
-            addJobStatusToStateStore(jobStatus, this.stateStore, this.eventProducer);
+            Optional<org.apache.gobblin.configuration.State> updatedJobStatus = addJobStatusToStateStore(jobStatus, this.stateStore);
+            boolean isJobStatusUpdated = updatedJobStatus.isPresent();
```

Comment:
I don't see a need for this intermediate name.  javadoc on `addJobStatusToStateStore` should be the place to convey semantics of `Optional::isPresent()` vs. not